markov/experimental.R

```diff
@@ -0,0 +1,431 @@
+
+pbp_first <- bind_rows(pbp_2020_first, pbp_2019_first, pbp_2018_first, 
+                       pbp_2017_first, pbp_2016_first) %>% 
+  arrange(desc(game_date))
+
+# Creates 1st inning RE24 tables for each season
+
+season_re24 <- function(pbp) {
+  
+  table <- pbp %>% 
+    run_expectancy_table() %>% 
+    mutate(on_1b = str_detect(base_out_state, "1b"),
+           on_2b = str_detect(base_out_state, "2b"),
+           on_3b = str_detect(base_out_state, "3b"),
+           outs = case_when(
+             str_detect(base_out_state, "0  outs") ~ 0,
+             str_detect(base_out_state, "1  outs") ~ 1,
+             str_detect(base_out_state, "2  outs") ~ 2
+           )) %>% 
+    select(outs, on_1b, on_2b, on_3b, base_out_state, avg_re)
+  
+  return(table)
+  
+}
+
+# Uses above function to get 1st inning RE24 tables for each year
+
+re24_2019_first <- season_re24(pbp_2019_first)
+re24_2018_first <- season_re24(pbp_2018_first)
+re24_2017_first <- season_re24(pbp_2017_first)
+re24_2016_first <- season_re24(pbp_2016_first)
+re24_2015_first <- season_re24(pbp_2015_first)
+
+# Tibble in which to collect game information
+# Getting the relevant game IDs before that
+
+game_ids <- pbp_2018_first %>%
+  select(game_pk) %>% 
+  unique() %>%
+  pull()
+
+game_info <- tibble(game_date = character(),
+                    game_id = numeric(),
+                    away_lineup = list(),
+                    pitcher_vs_away = numeric(),
+                    home_lineup = list(),
+                    pitcher_vs_home = numeric(),
+                    stadium = character(),
+                    team_home = character(),
+                    temperature = numeric())
+
+# Iterating over the selected games to grab info
+
+tracker <- 0
+
+for (game in game_ids) {
+  
+  tracker <- tracker + 1
+  print(tracker)
+  
+  game_info <- bind_rows(game_info, get_game_info(game))
+  
+}
+
+# Similar procedure as above, but now getting predictions instead
+
+result_data <- tibble(game_date = character(),
+                      game_id = numeric(),
+                      batter_id = numeric(),
+                      batter_name = character(),
+                      pitcher_id = numeric(),
+                      pitcher_name = character(),
+                      top_bot = character(),
+                      start = character(),
+                      end = character(),
+                      pred_woba = numeric(),
+                      pred_sing = numeric(),
+                      pred_re24 = numeric(),
+                      actual = numeric())
+
+# Function to get difference between Singlearity predictions and runs scored
+
+inning_diff <- function(game) {
+  
+  # Creates lineup using one call to markov_matrices(), including vars. like date
+  # Uses info we've already gathered
+  
+  indiv_game_info <- game_info %>% 
+    filter(game_id == game)
+
+  # Gets transition matrices for away team
+  
+  tmatrices_away <- markov_matrices(standard = FALSE,
```

Comment:
The next couple hundred lines of code look like they have a lot of duplicated code to handle home vs. away.  Get you avoid most of the code duplication?